### PR TITLE
CMR-4090 Deployment fixes for landing pages directory

### PR DESCRIPTION
**Pull request by oubiwann**
@chris-durbin provided some feedback after the initial deployment of the landing pages and sitemap features. Changes include:
* Alphabetical listing of providers
* Providers with no landing pages aren't linked
* All providers get a parenthetical count of landing pages after their link

Note that this branch is part 1 of 2; the second branch will include UX fixes for layout.

### CODE REVIEW

File: search-app/src/cmr/search/site/data.clj [hunk 1]
```diff
@@ -28,16 +28,39 @@
...
 
   Note that the given tag will be used to filter provider collections data
   that is used on the destination page."
-  [tag provider-data]
-  {:id (:provider-id provider-data)
-   :name (:provider-id provider-data)
-   :tag tag})
+  [context tag data]
+  (let [provider-id (:provider-id data)
+        provider-name provider-id
```
Comment (ygliuvt) [inline search-app/src/cmr/search/site/data.clj]: provider has a `short-name`. If we are putting a name out there, it probably should be the `short-name`.  I am ok with the html page just use the provider-id, then we can just drop `provider-name` from the data structure.

Comment (oubiwann) [inline search-app/src/cmr/search/site/data.clj]: Hrm...  this is a bit of code that's been overlooked -- thanks for the comment (setting `provider-name` to `provider-id` was a quick-fix originally done as a placeholder).

Comment (oubiwann) [inline search-app/src/cmr/search/site/data.clj]: Fixed.

**Comment by oubiwann** [comment]
Thanks, guys!

**Pull request merged**